cpp/pybind/data/data.cpp

```diff
@@ -0,0 +1,556 @@
+// ----------------------------------------------------------------------------
```

Comment:
`dataset.cpp`

Reply:
Since the module is `open3d::data`, not `open3d::dataset`, 
do you think, we can replace `dataset` by `data`. 
i.e. `cpp/open3d/data.cpp`, `cpp/open3d/data.h`, 
`data::Dataset -> data::Data`, `data::SimpleDataset`, `data::SingleDownloadData` ?